### DISCUSSION TITLE
File Organization (Folders)

### DISCUSSION THREAD

**Discussion by WhetDawg** [TortugaPower/BookPlayer#749] [Q&A]
BookPlayer appears to be everything that I'm needing as it keeps track of what I've listened to and where I left off at.
That said, I have 100s of purchased MP3s and I'm trying to find a way to organize them.
I've uploaded 24 files for my testing and I can tell it's going to be a mess unless I find a way to create the equivalent of file folders. 
I'm having no luck with this.
There are Playlists and the option to create a Volume (no idea what this is but it doesn't sound like it's what I want).
I've used the Files app and created a sub-folder under Processed but BookPlayer doesn't see it / show it.

Any suggestions?
Thanks!

**Comment by GianniCarlo** [comment]
Hi @WhetDawg ! yes, you should use the playlists as folders, we have enabled nested playlists too (to clarify, they're basically folders with autoplay turned on). Once you have your items inside folders/playlists, you can convert them into a single volume, so the app will treat them as a single book with multiple chapters.

Regarding the Processed folder, that actually is where we put the files after importing them from within the app (the app mantains a 1:1 parity between the digital library and the files on disk), so if you add or modify its contents, we don't have a listener to react to changes there, and it's only accessible for the user so the files can be exported or deleted manually (though both can be done via inside the app as well)

If you import multiple files at once, the app will prompt you to create a playlist or bind them into a single volume. To import files, you can do so from within the app too, it'll let you select multiple files via the integration we have to the Files app as well, or you can airdrop them. You can also put the files in the Documents folder (it's the one that contains the Processed folder), and the next time BookPlayer is launched (after force quitting it) it will prompt you with the files that it finds

**Reply by WhetDawg** [reply]
Thanks, Gianni.  I am loving this application!  The playlists work just like folders.  I checked again just now and I don't see anything on the main page or FAQ that explains what a volume.  What you posted here was very helpful.  Just sayin. 😉

It would be great if we could add imports to pulldown directly from the book developer's site. When I click "+ Add", I'd:

- chose book provider from a list
- enter my email address and a purchase key
- the files would download to the main folder or the corresponding playlist based on MP3 meta tag.

I found the Tip Jar and sent an "incredible" (your words) tip. Thank you all for developing this! 😁

**Reply by GianniCarlo** [reply]
Thanks a lot for your support 🙌!!, I'll update the FAQ with info about our volume feature. Regarding downloading, we do have a crude way of going about it with the [URL scheme download](https://github.com/TortugaPower/BookPlayer/wiki/URL-scheme#download), but it just applies to public files (unless the websites adds an access token in the url parameters). Other than that, I think Apple has some restrictions in their guidelines, and we'd have to adhere to some conditions before making it available in the UI (can't remember right now all the details)

**Comment by and-pete** [comment]
Didn't want to create a new Discussion for this, but just wanted to say a massive thanks for your continued work on this open app. I've just tipped as well :)

I'm particularly loving the new 'Combine Volume' feature.
It saves the headache of not knowing my progress when I import a group of MP3s into the app, only to see they don't have chapter numbers tagged/visible. 'Combine Volume' means a correct import order based off of the filenames seems to be sufficient. It's great.

Cheers again to the maintainers & your contributors :)
